### ISSUE TITLE
[Combat] Combat engine refactor

### ISSUE THREAD

**Issue by madsboddum** [ProjectSWGCore/Holocore#1347]
**NOTE**: #1346 should be completed first!

The combat engine code could really use a refactor.
Especially `CombatCommandAttack` - the class is huge and has a ton of complexity inside it. It has become difficult to expand.
The code is in this package: `com.projectswg.holocore.services.gameplay.combat.command`

The code is also hard to test, because there's a lot of chance involved. Perhaps when instrumenting the code, we can supply a hardcoded dice roll.

**Comment by Undercova** [comment]
is this still an "issue" / task ?

**Comment by madsboddum** [comment]
> is this still an "issue" / task ?

Yeah but probably a bit low priority for now. We have a lot of combat stuff implemented already, compared to many other game systems currently.
